### PR TITLE
Relocate spack_cmd and scripts from installer to root bin

### PR DESCRIPTION
**Pull request by johnwparent**
Remove superfluous copy operation needed to move spack batch launcher and associated scripts by refactoring spack to handle the invocation of the script from the root level bin directory. Refactor the installer to reflect this change.

### CODE REVIEW

**Comment by johnwparent** [comment]
@BetsyMcPhail Let me know if there is anyone else I should tag here.

File: lib/spack/docs/getting_started.rst [hunk 1]
```diff
@@ -1470,44 +1470,24 @@ Step 2: Install and setup Spack
...
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-To use Spack, run ``spack_cmd.bat`` (you may need to Run as Administrator).
-This will provide a Windows command prompt with an environment properly set
-up with Spack and its prerequisites. If you receive a warning message that
-Python is not in your ``PATH`` (which may happen if you installed Python
-from the website and not the Windows Store), add the location of the Python
-executable to your ``PATH`` now.
+To use Spack, run ``bin\spack_cmd.bat`` (you may need to Run as Administrator)
+from the spack soure tree root. This will provide a Windows command prompt
```
Comment (BetsyMcPhail) [inline lib/spack/docs/getting_started.rst]: typo 'spack source tree root`

**Review by BetsyMcPhail — CHANGES_REQUESTED** [review]
The shortcuts seem to be broken. My desktop shortcut, for example, is looking for `C:\Users\betsy.mcphail\Desktop\spack_cmd.bat` instead of the actual installed location.

Running directly from the bin directory works as expected

**Comment by johnwparent** [comment]
> The shortcuts seem to be broken. My desktop shortcut, for example, is looking for `C:\Users\betsy.mcphail\Desktop\spack_cmd.bat` instead of the actual installed location.
> 
> Running directly from the bin directory works as expected

How did the start menu shortcut work? Also to what directory did you attempt to install spack?

This behaved as expected locally, so it's most likely how I set the shortcut target w.r.t the install target file system.

**Comment by BetsyMcPhail** [comment]
> > The shortcuts seem to be broken. My desktop shortcut, for example, is looking for `C:\Users\betsy.mcphail\Desktop\spack_cmd.bat` instead of the actual installed location.
> > Running directly from the bin directory works as expected
> 
> How did the start menu shortcut work? Also to what directory did you attempt to install spack?
> 
> This behaved as expected locally, so it's most likely how I set the shortcut target w.r.t the install target file system.

The start menu shortcut is failing the same way. I installed into `C:\Spack 0.16.0_NEW1`

**Comment by johnwparent** [comment]
@BetsyMcPhail Thanks for the direction. It looks like the directory ids generated by CPack are not consistent enough to be referenced explicitly in the shortcut target. The new targets should be generic enough.  Sorry for the delay/incorrect targets.

**Comment by johnwparent** [comment]
@BetsyMcPhail whenever you get a chance, this update should work.

**Review by BetsyMcPhail — APPROVED** [review]

**Pull request merged**